### ISSUE TITLE
MIPS "Invalid register name global variable" when using sp in register variable

### ISSUE THREAD

**Issue by nathanchance** [llvm/llvm-project#54174]
The Linux kernel is looking to use `sp` in a register variable: https://git.kernel.org/mips/c/200ed341b8646ea893d5466974d6e107c46237ba

Unfortunately, this does not appear to be supported in `MipsTargetLowering::getRegisterByName()`, causing the following crash:

```
$ cat repro.c
register unsigned long current_stack_pointer __asm__("sp");

unsigned long foo(void)
{
        return current_stack_pointer;
}

$ clang --target=mipsel-linux-gnu -c -o /dev/null repro.c
fatal error: error in backend: Invalid register name global variable
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: clang --target=mipsel-linux-gnu -c -o /dev/null repro.c
1.      <eof> parser at end of file
2.      Code generation
3.      Running pass 'Function Pass Manager' on module 'repro.c'.
4.      Running pass 'MIPS DAG->DAG Pattern Instruction Selection' on function '@foo'
...
```

Is it possible to support this?

**Comment by nickdesaulniers** [comment]
https://reviews.llvm.org/D120926

**Comment by tstellar** [comment]
@atanasyan What do you think about backporting this? https://github.com/llvm/llvm-project/commit/e0adc3be132922776a867a623959e176f29f9965

**Comment by atanasyan** [comment]
@tstellar It's okay to backport this change.

**Comment by tstellar** [comment]
Merged: 65d53279b1fd 0826716786cd
